### PR TITLE
Association

### PR DESCRIPTION
**Pull request by wuletawwonte**
Changes in this PR
- Create Classroom class with label instance variable
- Create the `has-many/belongs-to` relationship between `Classroom` and `Student`
- Create a Class Book with title and author instance variables
- Create a class Rental with `date` instance variables
- Create the many-to-many relationship between Person and Book

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
```
Comment (danmainah) [inline book.rb:2]: - [ ] Kindly ensure you have `@rentals` should be present at your Getters and Setters(attr_accessor).

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Rental
+  attr_accessor :date
```
Comment (danmainah) [inline rental.rb:2]: - [ ] Kindly ensure you have `@book and @person` should be present at your Getters and Setters(attr_accessor).

File: person.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require './nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
```
Comment (danmainah) [inline person.rb:5]: - [ ] Kindly ensure you have `@rentals` should be present at your Getters and Setters(attr_accessor).

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @wuletawwonte, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 3 above)
Comment (wuletawwonte) [reply person.rb:5]: @danmainah

Comment (wuletawwonte) [reply person.rb:5]: ## Hello @danmainah 
Sorry, I have just one question. I have added custom setters for the instance variables(@rentals,@book, @person), I don't think I must add setters for those variables AGAIN. And, the linter(Rubocop) is not going to allow it either.

Comment (danmainah) [reply person.rb:5]: @wuletawwonte Kindly ask for a second opinion.

Comment (wuletawwonte) [reply person.rb:5]: @danmainah Sorry again, do you mean a second review request?

Comment (Gambit142) [reply person.rb:5]: @wuletawwonte, I understand that you have `getters` and not `setters` for the variables **_@rentals, @book, @person_**. Based on the project's requirements and the way the associations are set up, you should have setters for these variables. You can decide to add `attr_writer` for these variables or you remove the `attr_reader` and make it `attr_accessor`. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @wuletawwonte 👋, 

Great work attempting to understand the changes you were meant to implement by asking questions. 👏 

You've done well implementing some of the project's features, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
#### I am making this comment here because no change was made and I want to clear all that needs to be done at once.


- I understand that you have `getters` and not `setters` for the variables **_@rentals, @book, @person_**. Based on the project's requirements and the way the associations are set up, you should also have setters for these variables. You can decide to add `attr_writer` for these variables or you remove the `attr_reader` and make it `attr_accessor`. If you implement the changes this way, you won't encounter linters errors. 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Omar-Muhamad — APPROVED** [review]
# Approved ✔
### Hi @wuletawwonte,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 3 above)
Comment (Gambit142) [reply person.rb]: @wuletawwonte, you are welcome 👍

**Pull request merged**